src/sentry/api/endpoints/organization_incident_index.py

```diff
@@ -17,6 +28,41 @@ class OrganizationIncidentPermission(OrganizationPermission):
     }
 
 
+class IncidentSerializer(serializers.Serializer):
+    projects = ListField(
+        child=serializers.CharField(),
+        required=False,
+        allow_null=True,
+        default=[],
+    )
+    groups = ListField(
+        child=serializers.CharField(),
+        required=True,
+        allow_null=False,
+        default=[],
+    )
+    title = serializers.CharField(required=True)
+    query = serializers.CharField(required=True)
+    dateStarted = serializers.DateTimeField(required=True)
+    dateDetected = serializers.DateTimeField(required=False)
+
+    def validate_projects(self, attrs, source):
+        slugs = attrs[source]
+        projects = Project.objects.filter(slug__in=slugs)
```

Comment:
Should you also ensure that the projects are from the same org as the incident? I see that you do an access check below, but having an organization id condition here could help limit data loading in abuse scenarios.

Reply:
I'll do this for project, this data loading issue could still happen for groups though.